src/test/java/com/a2/backend/service/impl/TagServiceImplTest.java

```diff
@@ -0,0 +1,41 @@
+package com.a2.backend.service.impl;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import com.a2.backend.entity.Tag;
+import com.a2.backend.service.TagService;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.annotation.DirtiesContext;
+
+@SpringBootTest
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.ANY)
+class TagServiceImplTest {
```

Comment:
We should test the case "given and existing tag when the same tag is sent the old tag is reused and no additional tag is created"